docs: address README gaps (#106)
opensearch-project/opensearch-agent-server: README.md

## Prerequisites

- **Python 3.12+**
- **OpenSearch 2.x or 3.x** (local or remote cluster)

> - **AWS Bedrock** with access to a Claude inference profile.

Reviewer: While I think that testing was done mainly/exclusively with Anthropic's Claude model family I don't know for sure that only members of the Claude model family are supported

```suggestion
- **AWS Bedrock** with access to an inference profile.
```

> - **For Dashboards integration**: OpenSearch Dashboards **≥ 3.6**, matched to your OpenSearch major version. Matching exact versions avoids edge cases.

Reviewer: ```suggestion
- **For Dashboards integration**: OpenSearch Dashboards **≥ 3.6**, matching your OpenSearch major version. Matching exact versions avoids edge cases.
```

## Get Started

Pick the path that matches your situation:

| You want to... | Use this section |
|---|---|

> | Run the agent server against an existing OpenSearch cluster | [Option A — Install from PyPI](#option-a--install-from-pypi) |

Reviewer: ```suggestion
| Run the agent server and connect it to an existing OpenSearch cluster | [Option A — Install from PyPI](#option-a--install-from-pypi) |
```

| See everything (OS + OSD + agent + demo data) running end-to-end | [Option B — Full demo quickstart](#option-b--full-demo-quickstart) |

> | Hack on this repository's source code | [Option C — Run from source](#option-c--run-from-source) |

Reviewer: ```suggestion
| Contribute to this repository | [Option C — Run from source](#option-c--run-from-source) |
```

uiSettings:
  overrides:
    "home:useNewHomePage": true

# Sends current page context (app ID, filters, queries) to the agent.
contextProvider:
  enabled: true

# Chat -> agent server wiring.
chat:
  enabled: true
  agUiUrl: "http://localhost:8001/runs"
  # Forward the logged-in user's identity to the agent server via an
  # On-Behalf-Of token. Requires the OpenSearch security plugin with
  # OBO enabled (/_plugins/_security/api/obo/token). Leave false if
  # security is disabled; see "Credential forwarding" below.
  # forwardCredentials: true

# Workspaces (required for the agent's per-workspace data source routing).
# NOTE: workspaces are incompatible with the security plugin's multi-tenancy
# feature. If your cluster has opensearch_security.multitenancy.enabled=true,
# set it to false in opensearch.yml before enabling workspaces.
workspace.enabled: true
data_source.enabled: true
data_source.hideLocalCluster: false
```

**Start OpenSearch Dashboards:**

```bash
cd OpenSearch-Dashboards
yarn start --no-base-path
```

**Access the chat interface:**

- Open http://localhost:5601
- Click the chat icon in the top-right header
- Start chatting with your data

## Configuration

> The full list of configuration keys is in `.env.example`. The most common ones:

Reviewer: ```suggestion
The full list of configuration keys is in `.env.example`. The most common ones are:
```

```

### Chat icon doesn't appear in the OSD header

The chat button only renders when the new UI header is enabled:

```yaml
uiSettings:
  overrides:
    "home:useNewHomePage": true
```

Also confirm you're on OSD **≥ 3.6**; earlier versions ship an experimental chat plugin that doesn't integrate with this agent server.

### Workspaces don't work

Workspaces are incompatible with the security plugin's multi-tenancy feature. In your OpenSearch `opensearch.yml`:

```yaml
opensearch_security.multitenancy.enabled: false
```

Then restart OpenSearch. (Documented in OSD's own [explore plugin README](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/explore/README.md).)

### Agent says UBI data is unavailable

The ART agent's user-behavior-analysis sub-agent requires `ubi_queries` and `ubi_events` indices from the [UBI plugin](https://docs.opensearch.org/latest/search-plugins/ubi/). Without them, the agent reports that UBI data is unavailable and falls back to LLM-generated judgment lists for evaluation — other ART features (hypothesis generation, pairwise experiments, search configurations) still work.

> **Symptom to watch for:** the LLM may present an empty MCP tool result as a `502 Bad Gateway` message even though the underlying MCP call returned 200. If you don't need UBI features, use the **default agent** instead of the ART agent.

Reviewer: I don't understand the recommendation to use the default agent instead of the ART agent when UBI features are not available. When users don't have UBI data they can do everything except analysing user behavior by the agent. Why should they use an agent that doesn't know anything about relevance tuning for this task when there is no UBI data?
